src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,85 @@
-/* eslint-disable */
-import React from 'react';
+import { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { getFilteredTodos } from '../../utils/getFilteredTodos';
+
+export const TodoList = () => {
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const filter = useAppSelector(state => state.filter);
+  const dispatch = useAppDispatch();
+  const [filteredTodos, setFilteredTodos] = useState(todos);
+
+  const setCurrentTodo = (todo: Todo) => {
+    dispatch(currentTodoSlice.actions.setCurrentTodo(todo));
+  };
+
+  useEffect(() => {
+    setFilteredTodos(getFilteredTodos(todos, filter));
```

Comment:
Ensure that the `getFilteredTodos` utility function correctly handles the filtering logic based on the `filter` state. This is crucial for the correct display of filtered todos.